### PR TITLE
docs: add a note about the B2 download_url format

### PR DESCRIPTION
**Pull request by GGG-KILLER**
#### What is the purpose of this change?

Indicate how the B2 `download_url` configuration setting should be like.

#### Was the change discussed in an issue or in the forum before?

No.

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [ ] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/b2/b2.go [hunk 1]
```diff
@@ -160,7 +160,19 @@ free egress for data downloaded through the Cloudflare network.
 Rclone works with private buckets by sending an "Authorization" header.
 If the custom endpoint rewrites the requests for authentication,
 e.g., in Cloudflare Workers, this header needs to be handled properly.
-Leave blank if you want to use the endpoint provided by Backblaze.`,
+Leave blank if you want to use the endpoint provided by Backblaze.
+
+The URL provided here SHOULD have the protocol and SHOULD NOT have
+a trailing slash or specify the /file/bucket subpath rclone will
```
Comment (ncw) [inline backend/b2/b2.go:166]: Full stop after `subpath` maybe?

File: docs/content/b2.md [hunk 2]
```diff
@@ -466,6 +466,18 @@ If the custom endpoint rewrites the requests for authentication,
 e.g., in Cloudflare Workers, this header needs to be handled properly.
 Leave blank if you want to use the endpoint provided by Backblaze.
 
+The URL provided here SHOULD have the protocol and SHOULD NOT have
```
Comment (ncw) [inline docs/content/b2.md:469]: This section is auto generated from b2.go so you can lose this chunk.

File: backend/b2/b2.go [hunk 3]
```diff
@@ -160,7 +160,19 @@ free egress for data downloaded through the Cloudflare network.
...
+request files with "{download_url}/file/{bucket_name}/{path}".
+
+Example:
+Use https://mysubdomain.mydomain.tld instead of:
```
Comment (ncw) [inline backend/b2/b2.go:170]: I actually found this example quite confusing as it has a lot of wrong ways of doing it! I see what you were going for, but I'd rather just have the single correct example, or maybe

Example:

    https://mysubdomain.mydomain.tld

(No trailing `/`, `file` or `bucket`)

**Review by ncw — COMMENTED** [review]
Thanks for doing this :-)

See comments inline.

File: backend/b2/b2.go (hunk 1 above)
Comment (GGG-KILLER) [reply backend/b2/b2.go]: I went with "as" instead, thought it gives a bit more of continuity to the sentence.

What do you think?

File: docs/content/b2.md (hunk 2 above)
Comment (GGG-KILLER) [reply docs/content/b2.md]: Resolved

File: backend/b2/b2.go (hunk 3 above)
Comment (GGG-KILLER) [reply backend/b2/b2.go]: Resolved

**Review by ncw — APPROVED** [review]
That looks great - thank you - I'll merge that now.

**Pull request merged**